ledger/participant-integration-api/src/main/scala/platform/apiserver/configuration/LedgerConfigProvider.scala

```diff
@@ -21,25 +21,33 @@ object LedgerConfigProvider {
   )(implicit
       materializer: Materializer,
       loggingContext: LoggingContext,
-  ): ResourceOwner[CurrentLedgerConfiguration] =
+  ): ResourceOwner[CurrentLedgerConfiguration] = {
+    val scheduler = materializer.system.scheduler
     for {
       // First, we acquire the mechanism for looking up the current ledger configuration.
-      currentLedgerConfiguration <-
-        IndexStreamingCurrentLedgerConfiguration.owner(index, ledgerConfiguration)
+      currentLedgerConfiguration <- IndexStreamingCurrentLedgerConfiguration.owner(
+        ledgerConfiguration = ledgerConfiguration,
+        index = index,
+        scheduler = scheduler,
+        materializer = materializer,
+      )
       // Next, we provision the configuration if one does not already exist on the ledger.
       _ <- optWriteService match {
         case None => ResourceOwner.unit
         case Some(writeService) =>
           LedgerConfigProvisioner.owner(
+            ledgerConfiguration = ledgerConfiguration,
             currentLedgerConfiguration = currentLedgerConfiguration,
             writeService = writeService,
             timeProvider = timeProvider,
             submissionIdGenerator = SubmissionIdGenerator.Random,
-            ledgerConfiguration = ledgerConfiguration,
+            scheduler = scheduler,
+            executionContext = materializer.executionContext,
```

Comment:
Is this 100% safe? With time you never know where the execution context gets passed, and I think this represents an internal akka executor, which we should try not to abuse.

Reply:
Nope, I don't like this either. We know that the execution context is only used a little bit, but it could still interfere with Akka's own behavior. I'll pass the execution context designed for the API server through instead.

Reply:
Pushed.